### DISCUSSION TITLE
타입 선언 파일 확장자 변경 (.ts → .d.ts) 제안드립니당!

### DISCUSSION THREAD

**Discussion by yoonucho** [OttSideProject/OnDe-FE#16] [Ideas]
안녕하세요 !

현재 저희가 타입 선언을 .ts 파일에서 관리하고 있는데요, 이를 .d.ts 파일로 변경하는 것은 어떨까요?

## 1. 명확한 구분

d.ts는 타입 선언만 포함하므로, 해당 파일의 목적이 더 분명해집니다.

## 2. 컴파일 최적화

.d.ts 파일로 타입 선언을 분리하면, 타입 선언 파일이 컴파일 과정에서 실행 코드와 분리됩니다. 기존의 .ts 파일은 타입과 실행 코드를 모두 포함하기 때문에 TypeScript 컴파일러는 전체 파일을 처리해야 하지만, .d.ts 파일은 타입 선언만 포함되므로 컴파일 시 실제 JavaScript 코드로 변환되지 않고 타입 검사만을 수행하게 됩니다. 이는 컴파일러가 불필요한 코드를 처리하지 않아도 되기 때문에 컴파일 시간이 단축되고 프로젝트 전반의 컴파일 효율이 개선될 수 있습니다.

그리고 예원님이 고민하셨던  [Props의 타입 관리 방법](https://github.com/OttSideProject/OnDe-FE/pull/15#issuecomment-2385156058)
은 기존대로 컴포넌트 내부에 유지하는게 낫지 않을까 싶다는 생각이 드네요 제가 Props 타입은 컴포넌트 내부로 옮길게요! 
감사합니다.

**Comment by yewon830** [comment]
의견 주신 부분 좋은 거 같습니다! d.ts에 대해서 잘 몰랐는데 이번 기회로 학습하게 되었어요 감사합니다

**Comment by keemewoon** [comment]
확인했습니다 타입만 따로 관리하는것도 좋은것 같아요!
